Content.Client/Disposal/Visualizers/DisposalUnitVisualizerSystem.cs

```diff
@@ -1,104 +1,94 @@
 using Robust.Client.GameObjects;
-using Robust.Shared.Audio;
-using Robust.Shared.Serialization;
 using static Content.Shared.Disposal.Components.SharedDisposalUnitComponent;
 
 namespace Content.Client.Disposal.Visualizers
 {
-    [UsedImplicitly]
-    public sealed class DisposalUnitVisualizer : AppearanceVisualizer, ISerializationHooks
+    public sealed class DisposalUnitVisualizerSystem : VisualizerSystem<DisposalUnitComponent>
     {
+        [Dependency] private readonly SharedAudioSystem SoundSystem = default!;
         private const string AnimationKey = "disposal_unit_animation";
 
-        [DataField("state_anchored", required: true)]
-        private string? _stateAnchored;
-
-        [DataField("state_unanchored", required: true)]
-        private string? _stateUnAnchored;
-
-        [DataField("state_charging", required: true)]
-        private string? _stateCharging;
-
-        [DataField("overlay_charging", required: true)]
-        private string? _overlayCharging;
-
-        [DataField("overlay_ready", required: true)]
-        private string? _overlayReady;
-
-        [DataField("overlay_full", required: true)]
-        private string? _overlayFull;
-
-        [DataField("overlay_engaged", required: true)]
-        private string? _overlayEngaged;
-
-        [DataField("state_flush", required: true)]
-        private string? _stateFlush;
-
-        [DataField("flush_sound", required: true)]
-        private SoundSpecifier _flushSound = default!;
-
-        [DataField("flush_time", required: true)]
-        private float _flushTime;
+        public override void Initialize()
+        {
+            base.Initialize();
 
-        private Animation _flushAnimation = default!;
+            SubscribeLocalEvent<DisposalUnitComponent, ComponentInit>(OnComponentInit);
+        }
 
-        void ISerializationHooks.AfterDeserialization()
+        private void OnComponentInit(EntityUid uid, DisposalUnitComponent disposalUnit, ComponentInit args)
         {
-            _flushAnimation = new Animation {Length = TimeSpan.FromSeconds(_flushTime)};
+            disposalUnit.FlushAnimation = new Animation {
+                Length = TimeSpan.FromSeconds(disposalUnit.FlushTime)
+            };
 
             var flick = new AnimationTrackSpriteFlick();
-            _flushAnimation.AnimationTracks.Add(flick);
+            disposalUnit.FlushAnimation.AnimationTracks.Add(flick);
             flick.LayerKey = DisposalUnitVisualLayers.Base;
-            flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(_stateFlush, 0));
+            flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(disposalUnit.StateFlush, 0));
 
             var sound = new AnimationTrackPlaySound();
-            _flushAnimation.AnimationTracks.Add(sound);
+            disposalUnit.FlushAnimation.AnimationTracks.Add(sound);
+
+            sound.KeyFrames.Add(new AnimationTrackPlaySound.KeyFrame(SoundSystem.GetSound(disposalUnit.FlushSound), 0));
```

Comment:
It would be better if you made this a single object initializer

Reply:
It's a little awkward; AnimationsTrackSpriteFlick.KeyFrames has a protected setter, so you _have_ to go through the Add() interface. I've tidied up this code a bit, though.

Reply:
![image](https://user-images.githubusercontent.com/60196617/211178027-d18c82b7-4683-481a-836d-fbd0ea8df4f6.png)

I mean, that shouldnt really stop you from making  it a single object initializer

Reply:
I mean, you can still make it a single object initializer
![image](https://user-images.githubusercontent.com/60196617/211224193-36647d84-c86b-4cb6-bc77-9f5abca2c6e0.png)